### PR TITLE
Upgrade Rails workflow to true CI

### PR DESCRIPTION
**Pull request by indirect**
The existing Rails CI example only runs linters, which is not continuous integration. This change brings the Rails example workflow up to par with the other web framework CI flows, like Django.

This example is optimized for Rails 7, which does not include NodeJS, webpack, or yarn by default. No Rails application code changes are required for this flow to run the tests, and both minitest and rspec are supported via the `test` rake task.

### CODE REVIEW

File: ci/rubyonrails.yml
Comment (bishal-pdMSFT) [inline ci/rubyonrails.yml:35]: Should we also do `db:seed`?

Comment (indirect) [reply ci/rubyonrails.yml:35]: The `db:seed` task is generally not for generating test data. It is for setting up local development databases with sample data. See [this discussion with the Rails core team](https://discuss.rubyonrails.org/t/should-dbprepare-also-call-db-seed-by-default/74835) for more information.

File: ci/rubyonrails.yml
Comment (bishal-pdMSFT) [inline ci/rubyonrails.yml:25]: Specify these values as [environment variables](https://docs.github.com/en/actions/learn-github-actions/environment-variables) and use that across steps.

File: ci/rubyonrails.yml
Comment (bishal-pdMSFT) [inline ci/rubyonrails.yml:20]: Port can be environment variable as well

Comment (indirect) [reply ci/rubyonrails.yml:20]: Why should this be an environment variable? This is the default post used by Postgres on every machine in existence. Also, this port number is only used in one place, the DATABASE_URL below.

File: ci/properties/rubyonrails.properties.json
Comment (bishal-pdMSFT) [inline ci/properties/rubyonrails.properties.json:4]: This icon needs to be added to https://github.com/actions/starter-workflows/tree/main/icons directory.

File: ci/rubyonrails.yml
Comment (indirect) [reply ci/rubyonrails.yml:25]: Why should these be environment variables? The entire point is that they should not be configured. The database is configured with the exact values that are hardcoded in the test runner step.

If someone wants to configure their workflow to use different values, that is fine, they can do that! This is a template. But the point is that this action should work immediately, without making any changes to the user's system, and without making any changes to the Rails application code or configuration.

File: ci/properties/rubyonrails.properties.json
Comment (indirect) [reply ci/properties/rubyonrails.properties.json:4]: Done, my apologies. It seems I had a bad gitignore line that prevented me from adding the icon previously.

**Comment by indirect** [comment]
Okay, I've updated it to use environment variables. Just to warn you, I am sure it worked correctly before, because I tested it with my own Rails app. With the environment variable changes, I am no longer completely sure everything will work correctly. Hopefully it will, though!

File: ci/rubyonrails.yml
Comment (bishal-pdMSFT) [inline ci/rubyonrails.yml:35]: Please use the full SHA for any action which is not directly published by GitHub. See "general notes" section [here](https://github.com/actions/starter-workflows/blob/main/.github/pull_request_template.md).

Comment (indirect) [reply ci/rubyonrails.yml:35]: Done.

**Comment by indirect** [comment]
@bishal-pdMSFT it seems environment variables cannot be used for `services` as you requested. I tried it, and the workflow file is invalid: https://github.com/indirect/feedyour.email/actions/runs/1703936322/workflow#L21. There is also a forum discussion that seems to imply environment variables can only be used inside steps, which means it is impossible to share between the service and the step: https://github.community/t/how-to-use-env-context/16975/5.

**Comment by indirect** [comment]
@bishal-pdMSFT after testing the latest version that removes ENV duplication, it seems it is also impossible to refer to job env vars inside the job env var section. I have tested the workflow and resolved the error, but it seems it is not possible to arrange job steps and services to share data via environment variables. Here is the successful workflow run without the env var error, with a workflow matching the latest commit in this PR: https://github.com/indirect/feedyour.email/actions/runs/1703962976

**Comment by bishal-pdMSFT** [comment]
Thanks for pointing this out! The new changes look good to me.

**Comment by bishal-pdMSFT** [comment]
Though I will wait for either @aparna-ravindra or @anuragc617 to take a look before we merge.

File: ci/rubyonrails.yml
Comment (bishal-pdMSFT) [inline ci/rubyonrails.yml:33]: I do not see this commit SHA in any of the [releases](https://github.com/ruby/setup-ruby/releases) for this action. It would be good use a commit which has been officially released.

Comment (indirect) [reply ci/rubyonrails.yml:33]: Ok, updated to the SHA of v1.92.

File: ci/rubyonrails.yml
Comment (aparna-ravindra) [inline ci/rubyonrails.yml:49]: The same commit id used in the test job could be used here as well

File: ci/rubyonrails.yml
Comment (aparna-ravindra) [inline ci/rubyonrails.yml:18]: Do we need a health check to ensure that the service has started before it is used?

Comment (indirect) [reply ci/rubyonrails.yml:18]: I don't think so: I have run this workflow about 250 times now, and the alpine image starts up so fast I have never seen an error.

File: ci/rubyonrails.yml
Comment (indirect) [reply ci/rubyonrails.yml:49]: Ah, right. I will add it here as well.

**Comment by aparna-ravindra** [comment]
LGTM

**Review by bishal-pdMSFT — APPROVED** [review]

**Comment by bishal-pdMSFT** [comment]
@indirect let us know if you are fine for this to be merged?

**Comment by indirect** [comment]
@bishal-pdMSFT looks good, let’s do it 👍🏻

**Comment by bishal-pdMSFT** [comment]
@indirect merged. A huge thanks ❤️  for making starter workflows better 🎉

**Pull request merged**
